internal/controller/clusterworkloadserviceaccount_controller.go

```diff
@@ -64,51 +68,75 @@ func (r *ClusterWorkloadServiceAccountReconciler) Reconcile(
 		return ctrl.Result{}, err
 	}
 	if cwsa == nil {
+		log.V(1).Info("Resource not found", "name", req.Name)
 		return ctrl.Result{}, nil
 	}
 
+	cwsaResource := rules.NewClusterWSAResource(cwsa)
+
 	if isBeingDeleted(cwsa) {
-		cwsaResource := rules.NewClusterWSAResource(cwsa)
-		return handleDeletion(ctx, r.Client, req, cwsa, r.Engine, cwsaResource)
-	}
+		deleteEvent := &staging.EventInfo{
+			Resource:        cwsaResource,
+			EventType:       staging.EventTypeDelete,
+			Generation:      cwsa.GetGeneration(),
+			ResourceVersion: cwsa.GetResourceVersion(),
```

Comment:
I may be mistaken, but I'm pretty sure these are available on the data we put into the `WSAResource`.
Any reason not to add receiver methods to `WSAResource` and access them there?

Reply:
https://github.com/OctopusDeploy/octopus-permissions-controller/pull/55

Ended up capturing the values but simplified the construction.